### PR TITLE
Exemplar: revert wrong implementation.

### PR DESCRIPTION
**Pull request by songy23**
Updates https://github.com/census-instrumentation/opencensus-go/issues/1058.

Java counterpart: https://github.com/census-instrumentation/opencensus-java/pull/1791. The first step on refactoring exemplar APIs. More PRs will follow.

### CODE REVIEW

File: trace/exemplar.go [hunk 1]
```diff
@@ -18,14 +18,14 @@ import (
 	"context"
 	"encoding/hex"
 
-	"go.opencensus.io/exemplar"
+	"go.opencensus.io/metric/metricdata"
 )
 
 func init() {
-	exemplar.RegisterAttachmentExtractor(attachSpanContext)
+	metricdata.RegisterAttachmentExtractor(attachSpanContext)
```
Comment (bogdandrutu) [inline trace/exemplar.go]: We need to cleanup these things as well probably. Don't really understand why we need this.

File: tag/context.go [hunk 2]
```diff
@@ -45,10 +45,10 @@ type ctxKey struct{}
 var mapCtxKey = ctxKey{}
 
 func init() {
-	exemplar.RegisterAttachmentExtractor(extractTagsAttachments)
+	metricdata.RegisterAttachmentExtractor(extractTagsAttachments)
```
Comment (bogdandrutu) [inline tag/context.go]: This is a wrong usage of exemplars, tags are not "exemplars"

File: trace/exemplar.go (hunk 1 above)
Comment (songy23) [reply trace/exemplar.go]: Agree `trace` or `tag` shouldn't depend on `exemplar`. I removed `AttachmentExtractor` for now.

File: tag/context.go (hunk 2 above)
Comment (songy23) [reply tag/context.go]: Removed.

**Comment by rghetia** [comment]
Will this be a breaking change?

**Comment by songy23** [comment]
This is a breaking change to those who import "go.opencensus.io/exemplar". Though I don't believe any user would be using the previous exemplar APIs since no backend supports it before.

For `trace`, `tag`, `stats` and `metrics` packages, I don't think there's any breaking change.

**Comment by songy23** [comment]
This PR is only doing cleanups. More PRs will follow on refactoring the exemplar APIs.

**Comment by aviaoh** [comment]
@songy23 @rghetia 
This PR breaks `go get -v go.opencensus.io/exemplar` with the following error:
```
Fetching https://go.opencensus.io/exemplar?go-get=1
Parsing meta tags from https://go.opencensus.io/exemplar?go-get=1 (status code 200)
get "go.opencensus.io/exemplar": found meta tag get.metaImport{Prefix:"go.opencensus.io", VCS:"git", RepoRoot:"https://github.com/census-instrumentation/opencensus-go"} at https://go.opencensus.io/exemplar?go-get=1
get "go.opencensus.io/exemplar": verifying non-authoritative meta tag
Fetching https://go.opencensus.io?go-get=1
Parsing meta tags from https://go.opencensus.io?go-get=1 (status code 200)
go.opencensus.io (download)
package go.opencensus.io/exemplar: cannot find package "go.opencensus.io/exemplar" in any of:
	/usr/local/go/src/go.opencensus.io/exemplar (from $GOROOT)
	/Users/avia/dev/go/src/go.opencensus.io/exemplar (from $GOPATH)
```

This also breaks `go get github.com/Azure/go-autorest` @jhendrixMSFT

**Comment by aviaoh** [comment]
please see suggestion for solution in https://github.com/census-ecosystem/opencensus-go-exporter-ocagent/pull/49

**Comment by songy23** [comment]
Sorry for the trouble, I wasn't aware that there're downstream libraries depending on this. #1071 should fix it temporarily, meanwhile please migrate to the new package like https://github.com/census-ecosystem/opencensus-go-exporter-ocagent/pull/49.

**Pull request merged**